examples/dyn_phy.py

```diff
@@ -19,104 +21,133 @@
+    spec.set_grid(grid)
+
+    # set up domain decomposition
+    layout = spec.namelist.layout
+    partitioner = util.CubedSpherePartitioner(util.TilePartitioner(layout))
+    communicator = util.CubedSphereCommunicator(comm, partitioner)
+
+    # create a state from serialized data
+    savepoint_in = serializer.get_savepoint("FVDynamics-In")[0]
+    driver_object = fv3core.testing.TranslateFVDynamics([grid])
+    input_data = driver_object.collect_input_data(serializer, savepoint_in)
+    input_data["comm"] = communicator
+    state = driver_object.state_from_inputs(input_data)
+
+    # read in missing grid info for physics - this will be removed
+    dwind = TranslateUpdateDWindsPhys(grid)
+    missing_grid_info = dwind.collect_input_data(
+        serializer, serializer.get_savepoint("FVUpdatePhys-In")[0]
     )
-    step_physics(state)
-    if t % 5 == 0:
-        comm.Barrier()
-        output_vars = [
-            "u",
-            "v",
-            "ua",
-            "va",
-            "pt",
-            "delp",
-            "qvapor",
-            "qliquid",
-            "qice",
-            "qrain",
-            "qsnow",
-            "qgraupel",
-        ]
-        output = {}
-
-        for key in output_vars:
-            state[key].synchronize()
-            output[key] = np.asarray(state[key])
-        np.save("pace_output_t_" + str(t) + "_rank_" + str(rank) + ".npy", output)
 
+    # initialize dynamical core and physics objects
+    dycore = fv3core.DynamicalCore(
+        comm=communicator,
+        grid_data=spec.grid.grid_data,
+        grid_indexing=spec.grid.grid_indexing,
+        damping_coefficients=spec.grid.damping_coefficients,
+        config=spec.namelist.dynamical_core,
+        ak=state["atmosphere_hybrid_a_coordinate"],
+        bk=state["atmosphere_hybrid_b_coordinate"],
+        phis=state["surface_geopotential"],
+    )
+    step_physics = Physics(grid, spec.namelist, communicator, missing_grid_info)
+
+    print_for_rank0(f"Init & timestep 0 done in {time.time() - start}s ")
+
+    for t in range(1, int(time_steps)+1):
+        if rank == 0:
+            start = time.time()
+        dycore.step_dynamics(
+            state,
+            input_data["consv_te"],
+            input_data["do_adiabatic_init"],
+            input_data["bdt"],
+            input_data["ptop"],
+            input_data["n_split"],
+            input_data["ks"],
+        )
+        step_physics(state)
+        if t % 5 == 0:
+            io_start = 0
+            if rank == 0:
+                io_start = time.time()
+            comm.Barrier()
+            output_vars = [
+                "ua",
+                "qrain",
+            ]
+            output = {}
+
+            for key in output_vars:
+                state[key].synchronize()
+                output[key] = np.asarray(state[key])
+            np.savez_compressed(
+                "model_output/pace_output_t_" + str(t) + "_rank_" + str(rank) + ".npz",
```

Comment:
Do you need to create the model_output directory? Or will this automatically makes one?

Reply:
Done